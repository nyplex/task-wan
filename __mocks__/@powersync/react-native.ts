module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
