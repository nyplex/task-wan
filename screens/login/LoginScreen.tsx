import Animated, { useAnimatedStyle, interpolate } from "react-native-reanimated";
import { useReanimatedKeyboardAnimation } from "react-native-keyboard-controller";
import { Box } from "@/components/gluestack/box";
import Text from "@/components/Text";
import AppTitle from "@/components/UI/AppTitle";
import LoginThirdParty from "./components/LoginThirdParty";
import LoginDivider from "./components/LoginDivider";
import LoginFooter from "./components/LoginFooter";
import LoginForm from "./components/LoginForm";

const LoginScreen = () => {
  const { progress } = useReanimatedKeyboardAnimation();

  const animatedStyles = useAnimatedStyle(() => {
    const scale = interpolate(progress.value, [0, 1], [1, 0.01]);
    const height = interpolate(progress.value, [0, 1], [70, 5]);
    const opacity = interpolate(progress.value, [0, 1], [1, 1]);
    const marginBottom = interpolate(progress.value, [0, 1], [40, 0]);

    return {
      height,
      opacity,
      marginBottom,
      transform: [{ scale }],
    };
  });

  return (
    <Box className="flex-1 bg-backrgound pt-safe-offset-4 px-4">
      <Animated.View
        style={animatedStyles}
        className="overflow-hidden">
        <AppTitle />
      </Animated.View>

      <Box className="mt-12">
        <Text
          weight="semi-bold"
          className="text-center">
          Login to your account
        </Text>
      </Box>
      <LoginForm />
      <LoginDivider />
      <LoginThirdParty />
      <LoginFooter />
    </Box>
  );
};

export default LoginScreen;
