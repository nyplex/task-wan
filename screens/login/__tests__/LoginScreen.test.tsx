import { render } from "@testing-library/react-native";
import LoginScreen from "../LoginScreen";

// Mock all child components to isolate LoginScreen
const AppTitle = () => <></>;
AppTitle.displayName = "AppTitle";
const LoginForm = () => <></>;
LoginForm.displayName = "LoginForm";
const LoginDivider = () => <></>;
LoginDivider.displayName = "LoginDivider";
const LoginThirdParty = () => <></>;
LoginThirdParty.displayName = "LoginThirdParty";
const LoginFooter = () => <></>;
LoginFooter.displayName = "LoginFooter";

jest.mock("../components/LoginForm", () => LoginForm);
jest.mock("../components/LoginDivider", () => LoginDivider);
jest.mock("../components/LoginThirdParty", () => LoginThirdParty);
jest.mock("../components/LoginFooter", () => LoginFooter);
jest.mock("@/components/UI/AppTitle", () => AppTitle);
jest.mock("react-native-reanimated", () => require("react-native-reanimated/mock"));
jest.mock("react-native-keyboard-controller", () => ({
  useReanimatedKeyboardAnimation: () => ({ progress: { value: 0 } }),
}));

describe("LoginScreen", () => {
  it("renders without crashing", () => {
    render(<LoginScreen />);
    // No error thrown means pass
  });

  it("renders AppTitle, LoginForm, LoginDivider, LoginThirdParty, and LoginFooter", () => {
    const { UNSAFE_queryAllByType } = render(<LoginScreen />);
    expect(UNSAFE_queryAllByType(AppTitle).length).toBe(1);
    expect(UNSAFE_queryAllByType(LoginForm).length).toBe(1);
    expect(UNSAFE_queryAllByType(LoginDivider).length).toBe(1);
    expect(UNSAFE_queryAllByType(LoginThirdParty).length).toBe(1);
    expect(UNSAFE_queryAllByType(LoginFooter).length).toBe(1);
  });

  it("renders the login prompt text", () => {
    const { getByText } = render(<LoginScreen />);
    expect(getByText("Login to your account")).toBeTruthy();
  });

  it("matches the root structure snapshot", () => {
    const { getByTestId } = render(<LoginScreen />);
    const root = getByTestId("login-screen-root");
    expect(root.props.className).toContain("flex-1");
  });

  it("matches the rendered output snapshot", () => {
    const { toJSON } = render(<LoginScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
