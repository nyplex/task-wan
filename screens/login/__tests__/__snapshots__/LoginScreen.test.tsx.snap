// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen matches the rendered output snapshot 1`] = `
<View
  className="flex-1 bg-backrgound pt-safe-offset-4 px-4"
  testID="login-screen-root"
>
  <View
    className="overflow-hidden"
    style={
      {
        "height": undefined,
        "marginBottom": undefined,
        "opacity": undefined,
        "transform": [
          {
            "scale": undefined,
          },
        ],
      }
    }
  />
  <View
    className="mt-12"
  >
    <Text
      className="font-body based-styles font-semibold text-typography-primary text-[16px] tracking-[0px] text-center"
    >
      Login to your account
    </Text>
  </View>
</View>
`;
